#!/usr/bin/env python
from datetime import datetime
import pymysql
import time

class SaveTask():
    def __init__(self, task_id="", task_name="", run_time=""):
        self.db = pymysql.connect("10.10.83.175", "root", "root", "celery")
        self.task_name = task_name
        self.cursor = self.db.cursor()
        self.task_id = task_id
        self.run_time = run_time

    def update_task(self):
        sql = "update tasks set last_run_time = '"+self.run_time+"' where id = %d" %self.task_id
        try:
            self.cursor.execute(sql)
            self.db.commit()
        except:
            self.db.rollback()

    def save_task(self):
        dtime = time.strftime('%Y-%m-%d %H:%M:%S')
        sql = "INSERT INTO tasks(task_name, task_status, last_run_time, update_time) VALUES ('"+self.task_name+"', '1', '"+dtime+"', '"+dtime+"')"
        try:
            self.cursor.execute(sql)
            self.db.commit()
        except:
            self.db.rollback()

    def get_List(self):
        sql = "Select id,task_name,task_status,last_run_time,sec,crontab,task_type from tasks"
        try:
            self.cursor.execute(sql)
            # 获取所有记录列表
            results = self.cursor.fetchall()
            return list(results)
        except:
            print("Error: unable to fecth data")

    def init_task(self):
        init_time = (datetime.now()).strftime("%Y-%m-%d %H:%M:%S")
        sql = "update tasks set last_run_time = '"+init_time+"'"
        try:
            self.cursor.execute(sql)
            self.db.commit()
        except:
            self.db.rollback()

    def close_db(self):
        self.db.close()


